### PR TITLE
adding prerender support

### LINKED ISSUE #41: Prerendering

### PR DESCRIPTION
**Pull request by chenshay**
#41

### CODE REVIEW

File: src/viewer.js [hunk 1]
```diff
@@ -107,12 +111,18 @@ class Viewer {
...
     };
 
     if (this.referrer_) initParams.referrer = this.referrer_;
+    if (this.prerender_) {
+      initParams.visibilityState = 'prerender';
+      initParams.prerenderSize = 1;
+    }
+    if (this.prerender_ != undefined && !this.prerender_) {
+      initParams.prerenderSize = 0;
+    }
```
Comment (ericfs) [inline src/viewer.js]: Is it necessary to set prerenderSize to 0? If you're not prerendering does it matter what value it has?

File: src/viewer.js [hunk 2]
```diff
@@ -107,12 +111,18 @@ class Viewer {
...
 
-    // TODO (chenshay): set more init params.
     const initParams = {
       origin: parsedViewerUrl.origin,
     };
 
     if (this.referrer_) initParams.referrer = this.referrer_;
+    if (this.prerender_) {
+      initParams.visibilityState = 'prerender';
+      initParams.prerenderSize = 1;
```
Comment (ericfs) [inline src/viewer.js]: Do these need to be quoted so they're not obfuscated by the compiler? Or do you have externs?

Comment (chenshay) [inline src/viewer.js]: quotes. fixed

File: src/viewer.js (hunk 1 above)
Comment (chenshay) [inline src/viewer.js]: true. removed.

**Comment by ericfs** [comment]
LGTM, but I would recommend keeping #41 open until you also support switching from prerender to visible mode. Without that, you'd never use prerender.

**Pull request merged**